### PR TITLE
Import Sql Feature

### PR DESCRIPTION
**Pull request by harish81**
## Features Highlight
 - Import your existing database's table into generatedata's field set.
 - Paste table DDL SQL(Create table statement) or Select Database & Table to generate DDL.
 - Secure Database listing, Only work when you have enabled in your `settings.php` & logged in for login mode.
 - DB & Table listing Api Security.
 - Avoid SQL constraints.
 - Tested with Mysql, postgresql, oracle, sqlite.

## Demo
1. **Basic with paste SQL(Default Behaviour)**
new_generate_data_import_sql
2. **With List Database Features**
  > Enable `$importSqlListDatabases = true;` in `settings.php` file.

new_generate_data_import_sql2

## Modification  & Changes Explaination
 - **`library.php`** - to autoload `ImportSql.class.php` Class.
 - **`AjaxRequest.class.php`** - Add new 3 Api to Select Database, Table, And generate create SQL DDL.
 - **`Core.class.php`** - define 2 new property & their getter. 
      1. `$importSqlFeatureEnabled` for import sql feature enable/disable, 
      2. `$importSqlListDatabase` for database listing feature enable/disable.
 - **`ImportSql.class.php`** - A class to handle (List database, list table, create sql ddl) API.
 - **`Installation.class.php`** - Generate default set of configuration for Import Sql Feature. (By default list database is turned off).
 - **`en.php`** - Contain new language resource for import sql feature.
 - **`importSql.js`** - A New RequireJs Module to handle all import sql feature's stuff.
 - **`requireConfig.js`** - Define `importSql.js` Module.
 - **`generate.tab1.tpl`** - Add Import Sql UI & Dialog.
 -  **`templates/index.tpl`** - Load `importSql.js` Module.
 -  **`settings.example.php`** - Add Example Settings For Import Sql Feature.

> Sorry For typos, This Feature Will great for people to generate random data for their table. It will reduce overhead to type each field.

### CODE REVIEW

File: resources/templates/generate.tab1.tpl [hunk 1]
```diff
@@ -31,11 +31,43 @@
...
+			{/if}
 		</h2>
 
+		{if Core::isImportSqlFeatureEnabled()}
+		<div id="dialogSql" title="{$L.import_sql_input_placeholder}">
+			{if Core::isImportSqlListDatabase()}
+			<div>
+				<label>{$L.import_sql_select_db_tbl}</label>
+				<div>
+					<select name="m_sel_databases" id="m_sel_databases" style="width: 48%;">
```
Comment (benkeen) [inline resources/templates/generate.tab1.tpl]: Minor: could you use camel case for all field names and IDs? (`mSelDatabases`)

File: resources/scripts/importSql.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
+define([
```
Comment (benkeen) [inline resources/scripts/importSql.js]: Nice, thanks for making this its own module. Perfect.

File: library.php [hunk 3]
```diff
@@ -38,12 +38,12 @@
 require_once(__DIR__ . "/resources/classes/Templates.class.php");
 require_once(__DIR__ . "/resources/classes/Translations.class.php");
 require_once(__DIR__ . "/resources/classes/Utils.class.php");
+require_once(__DIR__ . "/resources/classes/ImportSql.class.php");
 
 // External libs
 require_once(__DIR__ . "/resources/libs/jsv4.php");
 require_once(__DIR__ . "/resources/libs/smarty/Smarty.class.php");
 require_once(__DIR__ . "/resources/classes/SecureSmarty.class.php");
-require_once(__DIR__ . "/resources/classes/ImportSql.class.php");
```
Comment (harish81) [inline library.php]: Move `ImportSql` class to custom libs

File: resources/classes/Core.class.php [hunk 4]
```diff
@@ -34,7 +34,7 @@ class Core {
 	private static $timeout = 300; // 5 minutes
 	private static $apiEnabled = false;
 	private static $importSqlFeatureEnabled=true;
-	private static $importSqlListDatabase=false;
+	private static $importSqlListDatabases=false;
```
Comment (harish81) [inline resources/classes/Core.class.php]: Rename `importSqlListDatabase` to `importSqlListDatabases`

File: resources/templates/generate.tab1.tpl [hunk 5]
```diff
@@ -41,14 +41,14 @@
...
 				<label>{$L.import_sql_select_db_tbl}</label>
 				<div>
-					<select name="m_sel_databases" id="m_sel_databases" style="width: 48%;">
+					<select name="mSelDatabases" id="mSelDatabases" style="width: 48%;">
 						<option value="" disabled>{$L.import_sql_select_db_option_hint}</option>
 					</select>
-					<select name="m_sel_table" id="m_sel_table" style="width: 48%;">
+					<select name="mSelTable" id="mSelTable" style="width: 48%;">
 						<option value="" disabled selected>{$L.import_sql_select_table_option_hint}</option>
 					</select>
```
Comment (harish81) [inline resources/templates/generate.tab1.tpl]: Rename fields to camelCase from snake_case

**Comment by benkeen** [comment]
I'll merge this in now, and include it in the next release. Note: I'm afraid it may be a few weeks; this next week will be very busy then I'm away for 10 days. But I'll get a new version out after.

Thanks again @harish81!

**Pull request merged**